docs/Using Fleet/GitOps.md

```diff
@@ -0,0 +1,548 @@
+  
+`default.yml` or `teams/team-name.yml`
+
+```yaml
+agent_options:
+  config:
+    decorators:
+      load:
+        - SELECT uuid AS host_uuid FROM system_info;
+        - SELECT hostname AS hostname FROM system_info;
+    options:
+      disable_distributed: false
+      distributed_interval: 10
+      distributed_plugin: tls
+      distributed_tls_max_attempts: 3
+      logger_tls_endpoint: /api/osquery/log
+      logger_tls_period: 10
+      pack_delimiter: /
+```
+
+##### Separate file
+ 
+`lib/agent-options.yml`
+
+```yaml
+config:
+  decorators:
+    load:
+      - SELECT uuid AS host_uuid FROM system_info;
+      - SELECT hostname AS hostname FROM system_info;
+  options:
+    disable_distributed: false
+    distributed_interval: 10
+    distributed_plugin: tls
+    distributed_tls_max_attempts: 3
+    logger_tls_endpoint: /api/osquery/log
+    logger_tls_period: 10
+    pack_delimiter: /
+```
+
+`default.yml` or `teams/team-name.yml`
+
+```yaml
+queries:
+  - path: ../lib/agent-options.yml
+# path is relative to default.yml or teams/team-name.yml
+```
+
+### controls
+
+The `controls` section allows you to configure scripts and device management (MDM) features in Fleet.
+
+- `scripts` is a list of paths to macOS, Windows, or Linux scripts.
+- `windows_enabled_and_configured` specifies whether or not to turn on Windows MDM features (default: `false`). Can only be configure for all teams (`default.yml`).
+- `enable_disk_encryption` specifies whether or not to enforce disk encryption on macOS and Widnows hosts (default: `false`).
+
+##### Example
+
+```yaml
+controls:
+  scripts: 
+    - path: ../lib/macos-script.sh 
+    - path: ../lib/windows-script.ps1
+    - path: ../lib/linux-script.sh
+  windows_enabled_and_configured: true
+  enable_disk_encryption: true # Available in Fleet Premium
+  macos_updates: # Available in Fleet Premium
+    deadline: "2023-06-13"
+    minimum_version: 13.4.1
+  windows_updates: # Available in Fleet Premium
+    deadline_days: 5
+    grace_period_days: 2
+  macos_settings:
+    - path: ../lib/macos-profile1.mobileconfig
+      labels:
+        - Label name 1
+    - path: ../lib/macos-profile2.json
+  windows_settings:
+    - path: ../lib/windows-profile.xml
+  macos_setup: # Available in Fleet Premium
+    bootstrap_package: https://example.org/bootstrap_package.pkg
+    enable_end_user_authentication: true
+    macos_setup_assistant: ../lib/dep-profile.json
+  macos_migration: # Available in Fleet Premium
+    enable: true
+    mode: voluntary
+    webhook_url: https://example.org/webhook_handler
+# paths are relative to default.yml or teams/team-name.yml 
+```
+
+#### macos_updates
+
+- `deadline` specifies the deadline in the form of `YYYY-MM-DD`. The exact deadline time is at 04:00:00 (UTC-8) (default: empty).
+- `minimum_version` specifies the minimum required macOS version (default: empty).
+
+#### windows_updates
+
+- `deadline_days` (default: empty)
+- `grace_period_days` (default: empty)
```

Comment:
Would also clarify the exact default values instead of "empty" (`0` or `null` I'd assume?)
```suggestion
- `deadline_days` (default: empty)
- `grace_period_days` (default: empty)
```

Reply:
I think I confirmed that the default value is `null` by looking at the API response in dev console:

![Screenshot 2024-07-01 at 9 41 41 AM](https://github.com/fleetdm/fleet/assets/47070608/7033f77a-e983-4d94-9d81-5acb31cdc3a6)

![Screenshot 2024-07-01 at 9 41 33 AM](https://github.com/fleetdm/fleet/assets/47070608/1df0d7d9-500a-47a6-85a6-d20f07f456ea)